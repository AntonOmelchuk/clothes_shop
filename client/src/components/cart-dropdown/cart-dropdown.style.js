import styled from 'styled-components';

export const CartDropdownContainer = styled.div`
  position: absolute;
  width: 240px;
  height: 340px;
  display: flex;
  flex-direction: column;
  padding: 20px;
  border: 1px solid black;
  background-color: white;
  top: 90px;
  right: 40px;
  z-index: 5;

  button {
    margin-top: auto;
  }

  button:hover {
    background: #fff;
  }
`;

export const CartItemsContainer = styled.div`
  height: 240px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  overflow-y: auto;
`;

export const EmptyCart = styled.span`
  font-size: 21px;
  margin: 90px auto;
`;
