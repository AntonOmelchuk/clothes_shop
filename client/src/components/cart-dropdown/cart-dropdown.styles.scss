.cart-dropdown {
  position: absolute;
  width: 240px;
  height: 340px;
  display: flex;
  flex-direction: column;
  padding: 20px;
  border: 1px solid black;
  background-color: white;
  top: 90px;
  right: 40px;
  z-index: 5;

  .cart-items {
    height: 240px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    overflow-y: auto;
  }

  .empty-cart {
    font-size: 21px;
    margin: 90px auto;
  }

  button {
    margin-top: auto;
  }
  button:hover {
    background: #fff;
  }
}
