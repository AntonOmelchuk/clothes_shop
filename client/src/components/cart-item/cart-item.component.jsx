import React from 'react';

import {
  CartItemContainer,
  ItemDetailsContainer,
  ItemImage,
} from './cart-item.style';

export const CartItem = ({item: {imageUrl, name, price, quantity}}) => {
  return (
    <CartItemContainer>
      <ItemImage src={imageUrl} alt={name} />
      <ItemDetailsContainer>
        <span>{name}</span>
        <span>
          {quantity} &times; ${price} ({price * quantity})
        </span>
      </ItemDetailsContainer>
    </CartItemContainer>
  );
};
