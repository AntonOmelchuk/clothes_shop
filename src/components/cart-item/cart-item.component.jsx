import React from 'react';

import './cart-item.styles.scss';

export const CartItem = ({item: {imageUrl, name, price, quantity}}) => {
  return (
    <div className='cart-item'>
      <img src={imageUrl} alt={name} />
      <div className='item-details'>
        <span className='name'>{name}</span>
        <span className='price'>
          {quantity} &times; ${price} ({price * quantity})
        </span>
      </div>
    </div>
  );
};
