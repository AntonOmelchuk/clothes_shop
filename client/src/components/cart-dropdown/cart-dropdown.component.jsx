import React from 'react';
import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';
import {createStructuredSelector} from 'reselect';

import {
  CartDropdownContainer,
  CartItemsContainer,
  EmptyCart,
} from './cart-dropdown.style';

import {CustomButton} from '../custom-button/custom-button.component';
import {CartItem} from '../cart-item/cart-item.component';

import {selectCartItems} from '../../redux/cart/cart.selectors';
import {actionClearCart, hideCart} from '../../redux/cart/cart.action';

export const CartDropdown = ({cartItems, history, dispatch}) => (
  <CartDropdownContainer>
    <CartItemsContainer>
      {cartItems.length ? (
        cartItems.map(cartItem => (
          <CartItem key={cartItem.id} item={cartItem} />
        ))
      ) : (
        <EmptyCart>Your cart is empty</EmptyCart>
      )}
    </CartItemsContainer>
    <CustomButton onClick={() => dispatch(actionClearCart())}>
      Clear cart
    </CustomButton>
    <CustomButton
      id='checkout-button'
      onClick={() => {
        history.push('/checkout');
        dispatch(hideCart());
      }}
    >
      Go to checkout
    </CustomButton>
  </CartDropdownContainer>
);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
});

export default withRouter(connect(mapStateToProps)(CartDropdown));
