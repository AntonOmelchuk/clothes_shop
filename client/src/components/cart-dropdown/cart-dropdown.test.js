import React from 'react';
import {shallow} from 'enzyme';

import {CartDropdown} from './cart-dropdown.component';
import {CartItem} from '../cart-item/cart-item.component';

let wrapper;
const mockCartItems = [{id: 1}, {id: 2}, {id: 3}];
const mockHistory = jest.fn();
const mockDispatch = jest.fn();

beforeEach(() => {
  const mockProps = {
    cartItems: mockCartItems,
    history: mockHistory,
    dispatch: mockDispatch,
  };

  wrapper = shallow(<CartDropdown {...mockProps} />);
});

it('renders CartDropdown component', () => {
  expect(wrapper).toMatchSnapshot();
});

it('renders an equal number CartItem components as the cartItems prop', () => {
  expect(wrapper.find(CartItem).length).toEqual(mockCartItems.length);
});
