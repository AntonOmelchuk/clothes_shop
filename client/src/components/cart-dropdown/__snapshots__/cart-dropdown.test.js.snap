// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CartDropdown component 1`] = `ShallowWrapper {}`;
